### PR TITLE
Blog app: add authorization rules

### PR DESCRIPTION
**Pull request by mustabbas**
- [ ] Installed CanCanCan in project.
- [ ] Added a role column to the users table. 
- [ ] Implemented A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Used CanCanCan for this authorization.
- [ ]  Implemented the post deleting functionality. Added the "Delete" button to the view and make sure that only authorized users can see it.
- [ ] Implemented A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
- [ ] Implemented the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -34,4 +34,10 @@ def create
       end
     end
   end
+
+  def destroy
+    Post.find(params[:id]).destroy
+    redirect_to all_posts_path(params[:user_id])
+    flash[:alert] = 'Your post was successfully deleted!'
+  end
```
Comment (AshakaE) [inline app/controllers/posts_controller.rb:44]: - [ ] Kindly set up `CanCanCan` for Post authorization

**Review by AshakaE — CHANGES_REQUESTED** [review]
### Hi @mustabbas  🙋‍♀️, 

Good job so far! 👌
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


#  Required Changes ♻️ 🚧
_Check the comments under the review._

###  Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (DyaryRaoof) [reply app/controllers/posts_controller.rb:44]: Hi. Could explain what you mean by that? we have already setup everything in the ability model

Comment (rloterh) [reply app/controllers/posts_controller.rb:44]: This issue is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Mustafa & Dyary, 

Your project is complete! Great job for making the necessary changes recommended in the previous review. There is nothing else to say other than... it's time to merge it 🍾 :shipit:. Congratulations! 🎉🥂

<br>
Cheers and Happy coding!👏 💻 🍷 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 1 more limited review per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**